charts/spire/charts/spire-server/templates/secret.yaml

```diff
@@ -101,10 +100,9 @@ telemetry:
 {{- end }}
 {{- end }}
 apiVersion: v1
-kind: ConfigMap
+kind: Secret
```

Comment:
Why are we switching this to a secret?

Can't we just put the secrets in a secret.

E.g.:

We could store the db creds in a secret and load them into the respective env var, this way we don't need to change the configmap:

See
http://www.postgresqlnow.com/connect-env-variables/

Reply:
To some extend it does protect yes, in the end k8s secrets are just obfuscated as base64.

In many cases we pull secrets from external sources injected as env variables, having the entire config as secret complicates this strategy.

Reply:
Users of k8s often confuse encoding and encryption or assume because one is used it means its designed to be obscured in favor over encryption. This is **not** the case in Kubernetes. base64 was chosen to encode secret data as its often binary data and needs to be put into a yaml/json document which isn't binary safe. But encryption is supported too. In many production systems, secrets are encrypted via:
https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/

The NSA's recommendation for Kubernetes hardening mentions it explicitly. I believe the DOD and CIS benchmark recommendations are also to turn on encryption for secrets.

My own clusters are encrypted. Secrets are often encrypted while configmaps are often not.

---

As for arbitrary plugin support, that's a debate on governance. I'm on the side of, if the spire folks needed plugin support to support the users, I'm for supporting the users in their needs as well. We'll hash that problem out in the governance meeting.

Reply:
Prefer to only have the password be a k8s secret

Reply:
Ok. I'll prepare another pr with that in it so we can review.

Struggling a bit with how all this reconciles with the principal of keeping things "simple". All this combined is the most complicated way of setting "connection_string" possible.

Reply:
Here it is. https://github.com/spiffe/helm-charts/pull/255